src/isl/util.rs

```diff
@@ -240,13 +242,44 @@ impl Range {
         })
     }
 
+    pub fn validate_precision_range(precision_range: &Range) -> IonSchemaResult<Range> {
```

Comment:
question—will this function catch the following edge cases (that are all valid for a `NonNegativeIntRange`)? Do we care?

1. `range::[min, 0]` – might be missed depending on whether the `min_value < max_value` check occurs before or after the `min` is substituted with `1`.
2. `range::[min, exclusive::1]` – `exclusive::1` is equivalent to `0` when used as the `max_value` of an integer range
3. `range::[exclusive::0, 10] – has a `0` as the lower bound, but is equivalent to `range::[1, 10]`, which is legal
4. `range::[1, max]` – equivalent to `range::[min, max]`, which is not allowed

I'm inclined to say that we need to make sure that (1) is covered. I'm not sure about (2). I think (3) is already illegal, and should remain that way, and that (4) is probably fine to allow.

Reply:
Yes, according to these concerns I feel like having a separate function to create precision ranges in `utils` module would make sense.

Reply:
With new changes https://github.com/amzn/ion-schema-rust/pull/83/commits/e2d3372c24973979609350b320a605c6fac01b88, now substituting min with 1 while parsing (previously it was doing this after range was parsed), which will cover all the above concerns. Right now for checking if `start < end` for any range is commented because no timestamp comparator for ion-rust is implemented. I will get latest version of ion-rust once its released to allow this check.